pkg/apis/app/v1beta1/kogitoinfra_types.go

```diff
@@ -60,28 +60,33 @@ type KogitoInfraSpec struct {
 	InfraProperties map[string]string `json:"infraProperties,omitempty"`
 }
 
+// RuntimeVariables defines the variables that will be
+// extracted from the linked resource and added to the
+// deployed Kogito service.
+type RuntimeVariables struct {
+	AppProps map[string]string `json:"appProps,omitempty"`
+	Env      []v1.EnvVar       `json:"env,omitempty"`
+}
+
+// RuntimeVariablesMap defines the map that KogitoInfraStatus
+// will use to link the runtime to their variables.
+type RuntimeVariablesMap map[RuntimeType]RuntimeVariables
+
 // KogitoInfraStatus defines the observed state of KogitoInfra.
 // +k8s:openapi-gen=true
 type KogitoInfraStatus struct {
 	Condition KogitoInfraCondition `json:"condition,omitempty"`
 
 	// +optional
-	// +mapType=atomic
-	// Application properties extracted from the linked resource that will be added to the deployed Kogito service.
-	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	AppProps map[string]string `json:"appProps,omitempty"`
-
-	// +optional
-	// +listType=atomic
-	// Environment variables extracted from the linked resource that will be added to the deployed Kogito service.
+	// Runtime variables extracted from the linked resource that will be added to the deployed Kogito service.
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	Env []v1.EnvVar `json:"env,omitempty"`
+	RuntimeVariables RuntimeVariablesMap `json:"runtimeVariables,omitempty"`
 
 	// +optional
 	// +listType=atomic
 	// List of volumes that should be added to the services bound to this infra instance
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
-	Volume []KogitoInfraVolume `json:"volumes,omitempty"`
+	Volume []KogitoInfraVolume `json:"volume,omitempty"`
```

Comment:
Why was that on changed ?

Reply:
I thought it was a mistake since it doesn't match the variable name. Should it be `volumes`?

Reply:
I guess we should avoid to modify the CRDs with breaking changes.
@ricardozanini @spolti wdyt ?

Reply:
@Kevin-Mok please keep it plural since it's an array. Also, we don't want to break the API since we've released 1.0 already.

Reply:
So, I remember the exact reason I changed it. When I run `./hack/go-vet.sh`, I get the following:
```
API rule violation: names_match,github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1beta1,KogitoInfraStatus,Volume
```
Changing it to `volume` fixes this. But, I guess I'll leave it anyway?

Reply:
Interesting, please rebase and I'll give it a go in your PR in my local env.